lib/network_manager.go

```diff
@@ -164,37 +166,17 @@ func (nm *NetworkManager) Start() {
 
 func (nm *NetworkManager) Stop() {
 	if !nm.params.DisableNetworkManagerRoutines {
-		nm.exitGroup.Add(4)
+		nm.exitGroup.Add(3)
 		close(nm.exitChan)
 		nm.exitGroup.Wait()
 	}
 	nm.DisconnectAll()
 }
 
-func (nm *NetworkManager) SetTargetOutboundPeers(numPeers uint32) {
-	nm.targetNonValidatorOutboundRemoteNodes = numPeers
-}
-
 // ###########################
 // ## NetworkManager Routines
 // ###########################
 
-// startPersistentConnector is responsible for ensuring that the node is connected to all persistent IP addresses. It
-// does this by periodically checking the persistentIpToRemoteNodeIdsMap, and connecting to any persistent IP addresses
-// that are not already connected.
-func (nm *NetworkManager) startPersistentConnector() {
```

Comment:
The routine to connect to persistent non-validators should not be a separate goroutine since it updates the same non-validator indices as the other non-validator connector routine